backend/de.metas.dunning/src/main/java/de/metas/dunning/api/impl/DunningBL.java

```diff
@@ -247,6 +252,75 @@ public void processCandidates(final IDunningContext context, final IDunningCandi
 		}
 
 		dunningProducer.finish();
+
+		sendMassNotifications(dunningProducer.getCreatedDunningDocIds(), Language.getBaseLanguage());
+	}
+
+	private void sendMassNotifications(@NonNull final Collection<DunningDocId> createdDunningDocIds,
+			@NonNull final Language language)
+	{
+		dunningDAO.getByIdsInTrx(createdDunningDocIds)
+				.stream()
+				.filter(doc -> groupByBoilerPlate(doc) != null)
+				.collect(Collectors.groupingBy(this::groupByBoilerPlate))
```

Comment:
intellij says: Function may return null, but it's not allowed here

Reply:
yeh, same as with Check.isBlank(String) not being considered as an upstream null-check, so is the filter above a null-check to prevent the error state that intellij is _seeing_.